apps/client/src/app/[user]/components/RecommendSection/index.tsx

```diff
@@ -1,23 +1,83 @@
+"use client";
+
+import SearchStudyInput from "@/app/[user]/components/SearchStudyInput";
+import { useSearchStudyQueryObject } from "@/app/api/groups/query";
+import imgEmpty from "@/asset/img/img_empty.png";
+import { useQuery } from "@tanstack/react-query";
+import Image from "next/image";
+import { useSearchParams } from "next/navigation";
 import RecommendCard from "./RecommendCard";
 import {
+  emptyGuideStyle,
+  emptyWrapper,
+  recommendHeaderContentWrapper,
   recommendHeaderWrapper,
   recommendSectionWrapper,
   recommendStudyTitle,
+  searchedStudyCountStyle,
+  studyListWrapper,
 } from "./index.css";
 
-const RecommendStudySection = () => (
-  <section
-    className={recommendSectionWrapper}
-    aria-labelledby="recommend-title"
-  >
-    <div className={recommendHeaderWrapper}>
-      <h2 id="recommend-title" className={recommendStudyTitle}>
-        추천 스터디
-      </h2>
-      <input />
-    </div>
-    <RecommendCard />
-  </section>
-);
+const RecommendStudySection = () => {
+  const searchParam = useSearchParams();
+  const searchPattern = searchParam.get("search");
+
+  const { data: studyList } = useQuery(
+    useSearchStudyQueryObject({ searchPattern: searchPattern || "" }),
+  );
+
+  return (
+    <section
+      className={recommendSectionWrapper}
+      aria-labelledby="recommend-title"
+    >
+      <div className={recommendHeaderWrapper}>
+        <div className={recommendHeaderContentWrapper}>
+          <h2 id="recommend-title" className={recommendStudyTitle}>
+            {searchPattern ? "검색 결과" : "추천 스터디"}
+          </h2>
+          {searchPattern && (
+            <div className={searchedStudyCountStyle}>
+              {studyList?.content.length || 0}
+            </div>
+          )}
+        </div>
+
+        <SearchStudyInput />
+      </div>
+
+      {searchPattern ? (
+        <ul className={studyListWrapper}>
+          {studyList?.content && studyList.content.length > 0 ? (
+            studyList.content.map((study) => (
+              <RecommendCard
+                key={study.id}
+                name={study.name}
+                introduction={study.introduction}
+                groupImage={study.groupImage}
+              />
+            ))
+          ) : (
+            <div className={emptyWrapper}>
+              <Image
+                src={imgEmpty}
+                alt="검색 결과가 없을 때 이미지"
+                width={369}
+                height={192}
+              />
+              <p className={emptyGuideStyle}>검색 결과가 없습니다.</p>
+            </div>
+          )}
+        </ul>
+      ) : (
+        <RecommendCard
+          name={"기본 스터디"}
+          introduction={"기본 스터디 입니다."}
+          groupImage={null}
+        />
+      )}
```

Comment:
중첩 분기는 보기 어려우니 컴포넌트를 분리해주는 게 좋겠어요.
먼저 list와 empty로 분리하고, list컴포넌트 안에서 empty를 early return 해주는 거죠!
그리고 여기엔 이렇게 바꿔주면 보기 좋을 거 같은데 어떠세요?
```suggestion
      {searchPattern ? (
        <SearchResult studyList={studyList} />
        ) : (
        <RecommendCard
          name={"기본 스터디"}
          introduction={"기본 스터디 입니다."}
          groupImage={null}
        />
      )}
```

Reply:
list컴포넌트 만들 때 `RecommendCard`를 `<li>`로 감싸주세요. ul의 자식 요소는 li만 와야 해요.

Reply:
전적으로 동의합니다.